WooCommerce/Classes/ViewRelated/JetpackSetup/JetpackSetupCoordinator.swift

```diff
@@ -119,26 +143,142 @@ private extension JetpackSetupCoordinator {
         benefitsController?.present(UINavigationController(rootViewController: viewController), animated: true)
     }
 
+    func startJetpackSetupFlow(authToken: String) {
+        /// Dismiss any existing login flow if possible.
+        if rootViewController.topmostPresentedViewController is LoginNavigationController {
+            rootViewController.dismiss(animated: false)
+        }
+        let progressView = InProgressViewController(viewProperties: .init(title: Localization.pleaseWait, message: ""))
+        rootViewController.topmostPresentedViewController.present(progressView, animated: true)
+        let action = JetpackConnectionAction.fetchJetpackUser { [weak self] result in
+            guard let self else { return }
+            self.rootViewController.dismiss(animated: true)
+            self.checkJetpackStatus(result, skipsWPComLogin: true)
+            #warning("TODO: sync account with token and start Jetpack setup")
+            DDLogInfo("✅ Ready for Jetpack setup - connection only: \(self.requiresConnectionOnly)")
+        }
+        stores.dispatch(action)
+    }
+}
+
+// MARK: - WPCom Login flow
+//
+private extension JetpackSetupCoordinator {
     func showWPComEmailLogin() {
         let emailLoginController = WPComEmailLoginHostingController(siteURL: site.url,
                                                                     requiresConnectionOnly: requiresConnectionOnly,
                                                                     onSubmit: checkWordPressComAccount(email:))
-        let loginNavigationController = UINavigationController(rootViewController: emailLoginController)
-        navigationController.dismiss(animated: true) {
-            self.navigationController.present(loginNavigationController, animated: true)
+        let loginNavigationController = LoginNavigationController(rootViewController: emailLoginController)
+        rootViewController.dismiss(animated: true) {
+            self.rootViewController.present(loginNavigationController, animated: true)
         }
         self.loginNavigationController = loginNavigationController
     }
 
     func checkWordPressComAccount(email: String) async {
         await withCheckedContinuation { continuation -> Void in
-            accountService.isPasswordlessAccount(username: email, success: { passwordless in
-                DDLogInfo("✅ account check done - passwordless: \(passwordless)")
-                continuation.resume()
-            }, failure: { error in
+            accountService.isPasswordlessAccount(username: email, success: { [weak self] passwordless in
+                self?.startAuthentication(email: email, isPasswordlessAccount: passwordless) {
+                    continuation.resume()
+                }
+            }, failure: { [weak self] error in
                 DDLogError("⛔️ Error checking for passwordless account: \(error)")
                 continuation.resume()
+                self?.handleAccountCheckError(error)
```

Comment:
I'm not sure what's the expected way to handle this either, because I got a comment the other day for not resuming `continuation` within the `guard let self` check 😅. I thought using `weak self` makes it safer here.

Reply:
maybe I highlighted the wrong diffs, I meant the success case where the continuation isn't triggered when `self` is deallocated. it's quite an edge case and looks pretty in code 🤷🏻‍♀️

Reply:
Ah I got it know 😅  Added a guard check in 3c5c1875857fa00a83a3709c1adc5dac7690f921.